Examples/Movies/android/app/src/main/java/com/facebook/react/movies/MoviesActivity.java

```diff
@@ -35,7 +36,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
     mReactInstanceManager = ReactInstanceManager.builder()
         .setApplication(getApplication())
-        .setBundleAssetName("MoviesApp.android.bundle")
+        .setJSBundleLoader(JSBundleLoader.createAssetLoader("MoviesApp.android.bundle"))
```

Comment:
I'm fine with allowing you to update the file on ReactInstanceManager (though like I said I don't think that's necessary for the reload case, but I could see other uses for it).

CatalystInstance/ReactBridge (Catalyst was the old name of RN if you didn't know) are single-use objects that correspond to a single script load and instantiation of RN. So when you reload JS, you actually get new instances of both that encapsulate the new JS/native module environment.